test/integration/acceptance/create_asset_test.cpp

```diff
@@ -0,0 +1,258 @@
+      .sendTx(complete(baseTx().createAsset(kAssetName, kDomain, kPrecision)))
+      .checkProposal(
+          [](auto &proposal) { ASSERT_EQ(proposal->transactions().size(), 1); })
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given a user with can_create_asset permission
+ * @when the user tries to create asset in valid but non existing domain
+ * @then stateful validation will be failed
+ */
+TEST_F(CreateAssetFixture, ValidNonExistingDomain) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTx(complete(
+          baseTx().createAsset(kAssetName, kNonExistingDomain, kPrecision)))
+      .checkProposal(
+          [](auto &proposal) { ASSERT_EQ(proposal->transactions().size(), 1); })
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 0); })
+      .done();
+}
+
+/**
+ * @given a user with can_create_asset permission
+ * @when the user tries to create an asset in a domain with illegal characters
+ * @then stateless validation failed
+ */
+TEST_F(CreateAssetFixture, InvalidDomain) {
+  IntegrationTestFramework itf(1);
+  itf.setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); });
+  for (const auto &domain : kIllegalDomainNames) {
+    itf.sendTx(complete(baseTx().createAsset(kAssetName, domain, kPrecision)),
+               checkStatelessInvalid);
+  }
+  itf.done();
+}
+
+/**
+ * C237 Create asset with a negative precision
+ * DISABLED because the current implementation of TransactionBuilder does not
+ * allow to pass negative value on a type level
+ * @given a user with can_create_asset permission
+ * @when the user tries to create an asset with negative precision
+ * @then stateless validation failed
+ */
+TEST_F(CreateAssetFixture, DISABLED_NegativePrecision) {
+  IntegrationTestFramework(1)
+      .setInitialState(kAdminKeypair)
+      .sendTx(makeUserWithPerms())
+      .skipProposal()
+      .checkBlock(
+          [](auto &block) { ASSERT_EQ(block->transactions().size(), 1); })
+      .sendTx(complete(baseTx().createAsset(kAssetName, kDomain, -10)),
+              checkStatelessInvalid)
+      .done();
+}
+
+/**
+ * C238 Create asset with overflow of precision data type
+ * DISABLED because the current implementation of TransactionBuilder does not
+ * allow to pass oversized value on a type level.
+ * @given a user with can_create_asset permission
+ * @when the user tries to create an asset with overflowed value of precision
+ * @then stateless validation failed
+ */
+TEST_F(CreateAssetFixture, DISABLED_PrecisionOverflow) {
+  uint64_t more_than_allowed = UCHAR_MAX + 1;
```

Comment:
`numeric_limits<PrecisionType>` might be more readable than `UCHAR_MAX`
https://en.cppreference.com/w/cpp/header/limits

Reply:
Discussed offline. Negative precision and overflow test cases will be removed because their meaningless were finally confirmed 🎉